### PR TITLE
Calico Networking K8s

### PR DESCRIPTION
**Pull request by YotamNordman**

### CODE REVIEW

File: k8s/calico-networking/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+
+
+In order to change this theres a parameter in ```variables.tf``` called  ```cni```.
+
+## Changing network encapsulation to bgp
+- change ```bgp``` parameter in ```variables.tf``` to true.
+A job will be created by rke to patch the requested change and this will attempt to rolling restart the daemonset of calico-node pods.
+Monitor that all the calico-node pods get restarted, help if required.
+- Install calicoctl in order to interact with calico
```
Comment (Slevy35) [inline k8s/calico-networking/README.md:20]: add link to documentation

File: k8s/calico-networking/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  ipipMode: Never
+  natOutgoing: false
+  nodeSelector: all()
+  vxlanMode: Never
+```
+If this is not the case we will edit the IPPool with kubectl and correct the parameters
+```
+kubectl edit ippool <ippoolname>
+```
+Again, Monitor the calico-node daemonset that should restart.
```
Comment (Slevy35) [inline k8s/calico-networking/README.md:50]:
```suggestion
> Important: Verify that all of the calico-node pods are restarting.
```

File: k8s/install/terraform/rke.tf [hunk 3]
```diff
@@ -545,3 +545,27 @@ resource "aws_lb_target_group_attachment" "rke-master-tg-servers-attachment" {
...
 # }
+resource "null_resource" "helm_config" {
+  count  = "${var.bgp == "true" ? 1 : 0}"
+  depends_on = [aws_lb_target_group_attachment.rke-master-tg-servers-attachment]
+
+  connection {
+    type        = "ssh"
+    user        = "centos"
+    private_key = file(local.private_key_file)
+    host        = aws_eip.bastion_eip.public_ip
```
Comment (Slevy35) [inline k8s/install/terraform/rke.tf:556]: use the bastion as jump server

File: k8s/install/terraform/rke.tf [hunk 4]
```diff
@@ -545,3 +545,27 @@ resource "aws_lb_target_group_attachment" "rke-master-tg-servers-attachment" {
 #   port              = 6443
 #   protocol          = "TCP"
 # }
+resource "null_resource" "helm_config" {
+  count  = "${var.bgp == "true" ? 1 : 0}"
```
Comment (Slevy35) [inline k8s/install/terraform/rke.tf:549]: fix bool

**Review by Slevy35 — CHANGES_REQUESTED** [review]

**Review by Slevy35 — CHANGES_REQUESTED** [review]

File: k8s/calico-networking/bgpconfiguration.yaml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+apiVersion: projectcalico.org/v3
+kind: BGPConfiguration
+metadata:
+  name: default
+spec:
+  asNumber: 64512
+  listenPort: 178
```
[the remark this answers is not in the document]
Comment (YotamNordman) [reply k8s/calico-networking/bgpconfiguration.yaml:7]: you're right, but it worked for us with 178 ill change it anyway

File: k8s/install/terraform/rke.tf [hunk 6]
```diff
@@ -545,3 +545,29 @@ resource "aws_lb_target_group_attachment" "rke-master-tg-servers-attachment" {
 #   port              = 6443
 #   protocol          = "TCP"
 # }
+resource "null_resource" "helm_config" {
+  count  = "${var.bgp == true ? var.rke_servers_count - 1 : 0}"
```
Comment (Slevy35) [inline k8s/install/terraform/rke.tf:549]: where is the seeder?

File: k8s/install/terraform/variables.tf [hunk 7]
```diff
@@ -54,4 +54,18 @@ variable "taint_servers" {
...
   description = "If set to false, user workloads would run on K8s master nodes"
-}
\ No newline at end of file
+}
+
+variable "cni" {
+  default     = "calico"
+  description = "cni options that rancher supports"
+}
+variable "bgp" {
```
Comment (Slevy35) [inline k8s/install/terraform/variables.tf:63]: change to `bgp_enabled`

**Review by Slevy35 — CHANGES_REQUESTED** [review]

**Review by Slevy35 — APPROVED** [review]

**Pull request merged**